Marlin/SanityCheck.h

```diff
@@ -430,11 +431,20 @@ static_assert(1 >= 0
  * Unified Bed Leveling
  */
 #if ENABLED(AUTO_BED_LEVELING_UBL)
-  #if ENABLED(DELTA)
-    #error "AUTO_BED_LEVELING_UBL does not yet support DELTA printers."
-  #elif DISABLED(NEWPANEL)
+  #if IS_KINEMATIC
+    #if ENABLED(DELTA)
+      #if DISABLED(UBL_DELTA)
+        #error "AUTO_BED_LEVELING_UBL requires UBL_DELTA for DELTA printers."
+      #endif
+    #else // SCARA
+      #error "AUTO_BED_LEVELING_UBL not supported for SCARA printers."
+    #endif
+  #endif
+  #if DISABLED(NEWPANEL)
     #error "AUTO_BED_LEVELING_UBL requires an LCD controller."
   #endif
+#elif ENABLED(UBL_DELTA)
+  #error "UBL_DELTA requires AUTO_BED_LEVELING_UBL."
```

Comment:
Since UBL_DELTA is set always when UBL and DELTA are set, this sanity check will never be invoked. Removing.